### PR TITLE
Tracked download

### PR DESCRIPTION
**Pull request by yohanboniface**
Includes:
- code refactoring for Sqlite database management
- code refactoring to allow overriding config (needed for secret info like email settings or secret key)
- new form to request a download with the IGN licence, storing user data in DB
- token management to allow only one download by request
- download URL (including token) sent by email
- some basic tests

Note: wordings are dummy and yet need to be done.

@noirbizarre r?
cc @davidbgk

### CODE REVIEW

File: adresse/templates/tracked_download_email.html [hunk 1]
```diff
@@ -0,0 +1,15 @@
+Bonjour {{ first_name }} {{ last_name }},
+
+Merci de blabla sous licence IGN.
```
Comment (davidbgk) [inline adresse/templates/tracked_download_email.html]: Huhu.

Comment (yohanboniface) [inline adresse/templates/tracked_download_email.html]: As said in the description of the PR, wordings are dummy ;)

**Comment by yohanboniface** [comment]
@davidbgk for the record, it was not possible to use the REFERER trick as we are in https ;)

File: adresse/static/main.css [hunk 2]
```diff
@@ -392,6 +413,32 @@ footer .twitter:before {
...
 }
+.flashes li {
+    padding: 10px;
+    text-align: center;
+}
+.flashes li:before,
+.flashes li:after {
+    font-family: 'open_sansextrabold';
+}
+.flashes li.success {
```
Comment (davidbgk) [inline adresse/static/main.css]: If you care about performances, it's probably better to remove the `li` selector here and below.

File: adresse/tracked_download.py [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+
+    def save(self):
+        DB.commit('INSERT INTO tracked_download '
+                  '(first_name, last_name, email, company, token) '
+                  'VALUES (?, ?, ?, ?, ?)',
+                  [self.first_name, self.last_name, self.email, self.company,
+                   self.token])
+
+    def use(self):
+        DB.commit('UPDATE tracked_download SET used=1 WHERE token=?',
```
Comment (davidbgk) [inline adresse/tracked_download.py]: You might want to increment the `used` column for each access to log attempts to reach files multiple times. This way you'll have data to arbitrate future discussions.

File: tests/test_tracked_download.py [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+        first_name='toto',
+        last_name='tata',
+        email='toto@tata.com',
+        company='toto SA',
+    )
+    dl.save()
+    url = url_for('download', token=dl.token)
+    resp = webapp.get(url, status=200)
+    assert resp.headers['X-Accel-Redirect'] == '/a/b/c/'
+    assert webapp.get(url, status=403)
```
Comment (davidbgk) [inline tests/test_tracked_download.py]: `test_cannot_xss`? :)

File: adresse/tracked_download.py [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+
+    def save(self):
+        DB.commit('INSERT INTO tracked_download '
+                  '(first_name, last_name, email, company, token) '
+                  'VALUES (?, ?, ?, ?, ?)',
+                  [self.first_name, self.last_name, self.email, self.company,
+                   self.token])
+
+    def use(self):
+        DB.commit('UPDATE tracked_download SET used=used+1 WHERE token=?',
```
Comment (davidbgk) [inline adresse/tracked_download.py]: AFAIK, it'll never be incremented given that you filter the token retrieval with `used=0` prior to `.use()` it ;)

Comment (yohanboniface) [inline adresse/tracked_download.py]: Yes :) But I thought that was a step forward anyway, so when I want to have an increment for read I just need to change on the view ;)

**Pull request merged**